src/librustc_session/options.rs

```diff
@@ -697,6 +708,8 @@ options! {CodegenOptions, CodegenSetter, basic_codegen_options,
         "compile the program with profiling instrumentation"),
     profile_use: Option<PathBuf> = (None, parse_opt_pathbuf, [TRACKED],
         "use the given `.profdata` file for profile-guided optimization"),
+    control_flow_guard: Option<CFGuard> = (None, parse_cfguard, [UNTRACKED],
+        "use Windows Control Flow Guard (`nochecks` or `checks`)"),
```

Comment:
This should probably start out as an unstable `-Z` flag (in `DebuggingOptions` below) instead of an insta-stable `-C` flag